src/chunk_dispatch_state.c

```diff
@@ -84,6 +84,7 @@ chunk_dispatch_exec(CustomScanState *node)
 
 		/* Find or create the insert state matching the point */
 		cis = ts_chunk_dispatch_get_chunk_insert_state(dispatch, point);
+		ts_chunk_insert_state_switch(cis, dispatch);
 
 		/*
 		 * Update the arbiter indexes for ON CONFLICT statements so that they
```

Comment:
Probably the code below needs modification as well. Always set the arbiter_index to cis->arbiter_indexes instead of doing this conditionally. Can we run into a case where we switch chunks and 1 has a arbiter index but not the other one?